src/main/java/it/gov/pagopa/payment/service/qrcode/QRCodeCreationServiceImpl.java

```diff
@@ -54,14 +62,19 @@ public TransactionResponse createTransaction(TransactionCreationRequest trxCreat
     TransactionInProgress trx =
         transactionCreationRequest2TransactionInProgressMapper.apply(trxCreationRequest, channel);
     trx.setMerchantId(merchantId);
+    trx.setAcquirerId(acquirerId);
+    trx.setIdTrxAcquirer(idTrxAcquirer);
```

Comment:
Maybe it should be better to move al these setters inside the mapper.
These are all mandatory set to perform in order to transform a TrxCreateRequest into a TrxInProgress...

Moving theme as parameter of the mapper would enforce their obligatoriness

Reply:
In order to keep the BiFunction implementation would be correct to pass a list of strings (channel + headers) to the mapper? Or we can ditch the BiFunction and just pass the single parameters?

Reply:
let's remove the BiFunction implementation ;)